statick_tool/plugins/reporting/write_jenkins_warnings_ng_reporting_plugin.py

```diff
@@ -0,0 +1,67 @@
+"""Write Statick results to Jenkins Warnings-NG plugin json-log compatible output."""
+from __future__ import print_function
+
+import json
+import os
+
+from statick_tool.reporting_plugin import ReportingPlugin
+
+
+class WriteJenkinsWarningsNGReportingPlugin(ReportingPlugin):
+    """Writes Statick results to Jenkins Warnings-NG plugin json-log compatible output."""
+
+    def get_name(self):
+        """Return the plugin name."""
+        return "write_jenkins_warnings_ng"
+
+    def report(self, package, issues, level):
+        """
+        Write the results to Jenkins Warnings-NG plugin compatible file.
+
+        Args:
+            package (:obj:`Package`): The Package object that was analyzed.
+            issues (:obj:`dict` of :obj:`str` to :obj:`Issue`): The issues
+                found by the Statick analysis, keyed by the tool that found
+                them.
+            level: (:obj:`str`): Name of the level used in the scan.
+        """
+        # Do not write report to file if no output directory is given.
+        if not self.plugin_context.args.output_directory:
+            return None, True
+
+        # We _should_ be in output_dir already, but let's be safe about it.
+        output_dir = os.path.join(self.plugin_context.args.output_directory,
+                                  package.name + "-" + level)
+
+        if not os.path.exists(output_dir):
+            os.mkdir(output_dir)
+        if not os.path.isdir(output_dir):
+            print("Unable to create output directory at {}!".format(
+                output_dir))
+            return None, False
+
+        output_file = os.path.join(output_dir,
+                                   package.name + "-" + level + ".statick")
```

Comment:
Should I change the filename here? Maybe add a `.json` before the `.statick`? Otherwise if both the regular and ng version of the plugin are enabled (which if no reporting plugins are specified is the case), then they'll overwrite each other's files.

Reply:
That makes sense to me.